### PR TITLE
use OS specific path separator

### PR DESCRIPTION
**Pull request by darkowlzz**
Replaces all the path creation lines with os.path.join to use os.sep.

### CODE REVIEW

File: operatorcourier/validate.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def _bundle_validation(self, bundle, repository=None):
                 validationDict[self.pkgsKey] = False
 
         valid = True
-        for key, value in validationDict.items():
+        for _, value in validationDict.items():
```
Comment (kevinrizza) [inline operatorcourier/validate.py]: While this seems like a valid thing to do, it seems unrelated to the purpose of this pr. Can you please remove it and create a separate pr for it? Or at least a separate commit and explain why this change is being made in a separate commit message?

Comment (darkowlzz) [reply operatorcourier/validate.py]: Sure, thanks.

File: operatorcourier/api.py [hunk 2]
```diff
@@ -111,7 +111,7 @@ def build_verify_and_push(namespace, repository, revision, token,
                               validation_output=validation_output, output=output)
 
     with TemporaryDirectory() as temp_dir:
-        with open('%s/bundle.yaml' % temp_dir, 'w') as outfile:
+        with open(os.path.join('%s', 'bundle.yaml') % temp_dir, 'w') as outfile:
```
Comment (jeremylinlin) [inline operatorcourier/api.py]:
```suggestion
        with open(os.path.join(temp_dir, 'bundle.yaml'), 'w') as outfile:
```

File: operatorcourier/push.py [hunk 3]
```diff
@@ -32,7 +33,7 @@ def push(self, bundle_dir, namespace, repository, release, auth_token):
 
     def _create_base64_bundle(self, bundle_dir, repository):
         with TemporaryDirectory() as temp_dir:
-            tarfile_name = "%s/%s.tar.gz" % (temp_dir, repository)
+            tarfile_name = os.path.join("%s", "%s.tar.gz") % (temp_dir, repository)
```
Comment (jeremylinlin) [inline operatorcourier/push.py]:
```suggestion
            tarfile_name = os.path.join(temp_dir, "%s.tar.gz") % repository
```

File: operatorcourier/nest.py [hunk 4]
```diff
@@ -38,7 +38,9 @@ def nest_bundles(yaml_files, registry_dir, temp_registry_dir):
 
     # write the package file
     package_name = package["packageName"]
-    with open('%s/%s.package.yaml' % (temp_registry_dir, package_name), 'w') as outfile:
+    packagefile_name = os.path.join('%s', '%s.package.yaml') % (
```
Comment (jeremylinlin) [inline operatorcourier/nest.py]:
```suggestion
    packagefile_name = os.path.join(temp_registry_dir, '%s.package.yaml') % package_name
```

File: operatorcourier/nest.py [hunk 5]
```diff
@@ -62,7 +64,8 @@ def nest_bundles(yaml_files, registry_dir, temp_registry_dir):
             crd_name = csv_crd["name"]
             if crd_name in crds:
                 crd = crds[crd_name]
-                with open('%s/%s.crd.yaml' % (csv_folder, crd_name), 'w') as outfile:
+                crdfile_name = os.path.join('%s', '%s.crd.yaml') % (csv_folder, crd_name)
```
Comment (jeremylinlin) [inline operatorcourier/nest.py]:
```suggestion
                crdfile_name = os.path.join(csv_folder, '%s.crd.yaml') % crd_name
```

**Comment by jeremylinlin** [comment]
Looks good to me overall. Just some formatting suggestions that may make code easier to read.

File: operatorcourier/nest.py (hunk 5 above)
Comment (MartinBasti) [reply operatorcourier/nest.py]: nope, this is correct
```python
os.path.join(csv_folder, '%s.crd.yaml' % crd_name')
```

File: operatorcourier/nest.py (hunk 4 above)
Comment (MartinBasti) [reply operatorcourier/nest.py]:
```python
os.path.join(temp_registry_dir, '%s.package.yaml' % package_name)
```

File: operatorcourier/nest.py (hunk 5 above)
Comment (MartinBasti) [reply operatorcourier/nest.py]: resp. both are correct, but way I mentioned seems more clear to me and less error prone

File: operatorcourier/validate.py (hunk 1 above)
Comment (MartinBasti) [reply operatorcourier/validate.py]:
```python
for value in validationDict.values():
```

File: operatorcourier/nest.py (hunk 5 above)
Comment (jeremylinlin) [reply operatorcourier/nest.py]: @MartinBasti Thanks for the suggestion!

**Comment by darkowlzz** [comment]
Thanks for all the suggestions. I've updated the two commits with the suggested changes.
